backend/drm/drm.c

```diff
@@ -1613,3 +1624,141 @@ void destroy_drm_connector(struct wlr_drm_connector *conn) {
 	wl_list_remove(&conn->link);
 	free(conn);
 }
+
+int wlr_drm_backend_get_read_only_fd(struct wlr_backend *backend) {
+	assert(backend);
+
+	struct wlr_drm_backend *drm = get_drm_backend_from_backend(backend);
+	char *path = drmGetDeviceNameFromFd2(drm->fd);
+	if (!path) {
+		wlr_log(WLR_ERROR, "Failed to get device name from DRM fd");
+		return -1;
+	}
+
+	int fd = open(path, O_RDWR);
+	if (fd < 0) {
+		wlr_log_errno(WLR_ERROR, "Unable to clone DRM fd for client fd");
+		free(path);
+		return -1;
+	}
+
+	if (drmIsMaster(fd) && drmDropMaster(fd) < 0) {
+		wlr_log_errno(WLR_ERROR, "Failed to drop master");
+		return -1;
+	}
+
+	return fd;
+}
+
+int wlr_drm_create_lease(struct wlr_output **outputs, size_t n_outputs,
```

Comment:
I think @subdiff has mentioned this before: maybe it would be a good idea to return a `struct wlr_drm_lease *` here? That way, we could provide a `destroy` event on it, that we can fire when the kernel notifies us via a uevent that the lease has been destroyed.

Reply:
That could be an idea. The `wlr_drm_lease` struct will only hold the leased fd and the event?

Reply:
It would be nicer for the FD to not be part of the struct, so that the caller can `close` it early in case it sends it to a Wayland client.

The struct would only contain the event and the lease ID for now, I think. Then `wlr_drm_revoke_lease` can take the struct as an argument.

TBH, I wouldn't even mind not implementing the destroy event for now, keep that as a TODO. A spurious ENOENT isn't that of a big deal.

Reply:
I'll slap a TODO then.